velox/functions/prestosql/aggregates/tests/ArbitraryTest.cpp

```diff
@@ -349,6 +349,60 @@ TEST_F(ArbitraryTest, interval) {
   testAggregations({data}, {}, {"arbitrary(c2)"}, "SELECT null");
 }
 
+TEST_F(ArbitraryTest, longDecimal) {
+  auto data = makeRowVector({// Grouping key.
+                             makeFlatVector<int64_t>({1, 1, 2, 2, 3, 3, 4, 4}),
+                             makeNullableFlatVector<int128_t>(
+                                 {HugeInt::build(10, 100),
+                                  HugeInt::build(10, 100),
+                                  HugeInt::build(10, 200),
+                                  HugeInt::build(10, 200),
+                                  std::nullopt,
+                                  std::nullopt,
+                                  std::nullopt,
+                                  HugeInt::build(10, 400)},
+                                 DECIMAL(38, 8))});
+
+  auto expectedResult = makeRowVector({
+      makeFlatVector<int64_t>({1, 2, 3, 4}),
+      makeNullableFlatVector<int128_t>(
+          {HugeInt::build(10, 100),
+           HugeInt::build(10, 200),
+           std::nullopt,
```

Comment:
From presto doc:
```
arbitrary(x) → [same as input][#](https://prestodb.io/docs/current/functions/aggregate.html#arbitrary)
Returns an arbitrary non-null value of x, if one exists.
```
Would you please verify this behavior is consistent with Presto JAVA.

Reply:
The behavior is indeed consistent with Presto JAVA.

Presto java:
```
presto:sf1> SELECT arbitrary(Col) FROM (VALUES cast(0.82 as decimal(5,4)), cast(2.333 as decimal(5,4)), cast(3.132 as decimal(5,4)), cast(4.344 as decimal(38,4))) AS X(Col);
 _col0  
--------
 0.8200 
(1 row)

presto:sf1> select arbitrary(cast(3.3 as decimal(38,4)));
 _col0  
--------
 3.3000 
(1 row)
```

Velox c++:
```
presto:tpch> SELECT arbitrary(Col) FROM (VALUES cast(0.82 as decimal(5,4)), cast(2.333 as decimal(5,4)), cast(3.132 as decimal(5,4)), cast(4.344 as decimal(38,4))) AS X(Col);
 _col0  
--------
 0.8200 
(1 row)

presto:tpch> select arbitrary(cast(3.3 as decimal(38,4)));
 _col0  
--------
 3.3000 
(1 row)
```

Reply:
Sorry, I should have been more specific. Validate this test case in JAVA:

```
  makeFlatVector<int64_t>({1, 1, 2, 2, 3, 3, 4, 4}),
                             makeNullableFlatVector<int128_t>(
                                 {HugeInt::build(10, 100),
                                  HugeInt::build(10, 100),
                                  HugeInt::build(10, 200),
                                  HugeInt::build(10, 200),
                                  std::nullopt,
                                  std::nullopt,
                                  std::nullopt,
                                  HugeInt::build(10, 400)},
                                 DECIMAL(38, 8))});
```

Reply:
Validated the testcase, both Presto Java and Velox C++ are behaving the same:

Presto Java:
```
presto> SELECT arbitrary(value), key 
     -> FROM 
     -> ( VALUES 
     -> (1, cast(cast(100 as BIGINT) as decimal(38,8))), (1, cast(cast(100 as BIGINT) as decimal(38,8))), (2, cast(cast(200 as BIGINT) as decimal(38,8))), (2, cast(cast(200 as BIGINT) as decimal(38,8))), (3, null), (3, null), (4, cast(cast(400 as BIGINT) as decimal(38,8))), (4, cast(cast(400 as BIGINT) as decimal(38,8))) ) 
     -> AS 
     -> t(key, value) group by key;
    _col0     | key 
--------------+-----
 NULL         |   3 
 200.00000000 |   2 
 400.00000000 |   4 
 100.00000000 |   1 
(4 rows)
```

Velox C++:
```
presto:tpch> SELECT arbitrary(value), key 
          -> FROM 
          -> ( VALUES 
          -> (1, cast(cast(100 as BIGINT) as decimal(38,8))), (1, cast(cast(100 as BIGINT) as decimal(38,8))), (2, cast(cast(200 as BIGINT) as decimal(38,8))), (2, cast(cast(200 as BIGINT) as decimal(38,8))), (3, null), (3, null), (4, cast(cast(400 as BIGINT) as decimal(38,8))), (4, cast(cast(400 as BIGINT) as decimal(38,8))) ) 
          -> AS 
          -> t(key, value) group by key;
    _col0     | key 
--------------+-----
 400.00000000 |   4 
 100.00000000 |   1 
 NULL         |   3 
 200.00000000 |   2 
(4 rows)
```